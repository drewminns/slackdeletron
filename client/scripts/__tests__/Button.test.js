import React from 'react';
import { shallow, mount } from 'enzyme';

import Button from '../Components/Button';

describe('<Button />', () => {
  it('Mounts a Button element correctly', () => {
    const button = shallow(<Button />);
    expect(button).toHaveLength(1);
  });

  it('Displays the correct text when provided as a prop', () => {
    const TEXT = 'CLICK ME';
    const button = mount(<Button text={TEXT} />);
    expect(button.text()).toBe(TEXT);
  });

  it('Renders a button element by default', () => {
    const TEXT = 'CLICK ME';
    const button = mount(<Button text={TEXT} />);
    expect(button.find('button')).toHaveLength(1);
    expect(button.find('a')).toHaveLength(0);
  });

  it('Renders an A element when required', () => {
    const TEXT = 'CLICK ME';
    const button = mount(<Button text={TEXT} isLink href="#" />);
    expect(button.find('button')).toHaveLength(0);
    expect(button.find('a')).toHaveLength(1);
  });

  it('Does not render an A element if href prop is not provided', () => {
    const TEXT = 'CLICK ME';
    const button = mount(<Button text={TEXT} isLink />);
    expect(button.find('button')).toHaveLength(1);
    expect(button.find('a')).toHaveLength(0);
  });
});
