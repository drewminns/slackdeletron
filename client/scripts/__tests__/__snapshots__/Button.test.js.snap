// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Renders Correctly 1`] = `
<button
  className="sc-bdVaJa bnvqMO"
  onClick={[Function]}
>
  
</button>
`;
